### PR TITLE
📝 Add Formik base

### PR DESCRIPTION
**Pull request by Qrtn**
## Status:
:construction: In development

## Description

Fixes #<number>

### CODE REVIEW

File: src/components/form/ReportModal.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    // Should width be manually adjusted?
+    width: 130px;
+
+    &:not(last-child) {
+      margin-right: 12px;
+    }
+  }
+`
+
+const ReportModal = ({ typeId: _typeId, name, onDismiss, ...props }) => (
```
Comment (SirajChokshi) [inline src/components/form/ReportModal.js:38]: will we ever use `_typeId`?

File: src/components/map/MapPage.js [hunk 2]
```diff
@@ -18,14 +20,17 @@ import Map from './Map'
  */
 const VISIBLE_CLUSTER_ZOOM_LIMIT = 12
 
+const ADD_LOCATION_ZOOM = 18
```
Comment (SirajChokshi) [inline src/components/map/MapPage.js:23]: Comment for this const?

File: src/components/ui/AddLocationPin.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg width="44" height="57" viewBox="0 0 44 57" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (SirajChokshi) [inline src/components/ui/AddLocationPin.svg:1]: is this not available in a library?

File: src/components/ui/Modal.js [hunk 4]
```diff
@@ -3,11 +3,12 @@ import '@reach/dialog/styles.css'
...
 
+// TODO: fix vertical centering
+
 const Modal = styled(Dialog)`
   border-radius: 23px;
-  padding: 23px 17px 23px 17px;
-  width: 75%;
-  height: 70%;
+  padding: 23px 17px;
+  width: 80%;
```
Comment (SirajChokshi) [inline src/components/ui/Modal.js:11]: Consider adding a min-width for some cases

File: src/components/ui/ResetButton.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ const ResetButton = styled.button`
   padding: 0;
   font-family: inherit;
   cursor: pointer;
-  outline: inherit;
+  // outline: inherit;
```
Comment (SirajChokshi) [inline src/components/ui/ResetButton.js:9]: why is comment here?

**Review by SirajChokshi — COMMENTED** [review]
Good work! `src/components/form/FormContext.js` is an empty file, I would look into that. Approved since this is generally working 🚀

File: src/components/form/ReportModal.js (hunk 1 above)
Comment (Qrtn) [reply src/components/form/ReportModal.js:38]: Yes, once reporting is enabled.

File: src/components/form/FormikStepper.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  }
+`
+
+const ProgressButtons = styled.div`
+  margin-bottom: 16px;
+  text-align: center;
+
+  button {
+    // Should width be manually adjusted?
+    width: 130px;
```
Comment (code-arman) [inline src/components/form/FormikStepper.js:32]: i assume this is a @SirajChokshi question?

**Review by SirajChokshi — APPROVED** [review]
Great work!

File: src/components/form/FormikStepper.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  const currentChild = childrenArray[step]
+  const isLastStep = step === childrenArray.length - 1
+
+  const handleSubmit = async (values, helpers) => {
+    if (isLastStep) {
+      await onSubmit(values, helpers)
+      setCompleted(true)
+    } else {
+      setStep((s) => s + 1)
+      // helpers.setTouched({})
```
Comment (code-arman) [inline src/components/form/FormikStepper.js:56]: i assume this comment is left here intentionally for later use?

File: src/components/form/FormikStepper.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    <Formik
+      {...props}
+      validationSchema={currentChild.props.validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({ isSubmitting }) => (
+        <StyledForm>
+          <div>{currentChild}</div>
+
+          <ProgressButtons $twoButtons>
```
Comment (code-arman) [inline src/components/form/FormikStepper.js:70]: $fancy$

File: src/components/form/LocationForm.js [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+      <Select name="season_start" options={MONTH_OPTIONS} />
+      <span>to</span>
+      <Select name="season_end" options={MONTH_OPTIONS} />
+    </InlineSelects>
+  </>
+)
+
+const Step2 = () => {
+  const fileUploadRef = useRef()
+  // TODO: instead of doing this... just wrap both the file upload and the caption inputs in a new Formik field
```
Comment (code-arman) [inline src/components/form/LocationForm.js:112]: I assume this is for another PR?

File: src/components/form/LocationForm.js [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+export const LocationForm = () => {
+  const [typeOptions, setTypeOptions] = useState([])
+
+  useEffect(() => {
+    async function fetchTypes() {
+      const types = await getTypes()
+      const options = types.map((t) => ({
+        value: t.id,
+        label: t.name,
+      }))
```
Comment (code-arman) [inline src/components/form/LocationForm.js:187]: How about `type` instead of `t`?

File: src/components/form/ReportModal.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  }
+`
+
+const Buttons = styled.div`
+  text-align: center;
+  margin-top: 16px;
+
+  button {
+    // Should width be manually adjusted?
+    width: 130px;
```
Comment (code-arman) [inline src/components/form/ReportModal.js:30]: Siraj question again?

File: src/components/ui/AddLocationPin.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components/macro'
+
+import PinSvg from './AddLocationPin.svg'
+
+// TODO: use icon from a library
```
Comment (code-arman) [inline src/components/ui/AddLocationPin.js:5]: future PR?

File: src/components/ui/AddLocationPin.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+const AddLocationPin = styled.img.attrs({
+  src: PinSvg,
+})`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  // How to do this without transform/should I?
+  transform: translate(-50%, -100%);
```
Comment (code-arman) [inline src/components/ui/AddLocationPin.js:14]: Siraj question i assume?

File: src/components/ui/AddLocationPin.svg (hunk 3 above)
Comment (Qrtn) [reply src/components/ui/AddLocationPin.svg:1]: Added a TODO

File: src/components/ui/Select.js [hunk 14]
```diff
@@ -17,7 +19,7 @@ const StyledSelect = styled(Select)`
   }
 
   .select__control {
-    border: 1px solid ${({ theme }) => theme.secondaryBackground};
+    border: 1px solid ${validatedColor()};
```
Comment (code-arman) [inline src/components/ui/Select.js:22]: nice

File: src/components/ui/Slider.js [hunk 15]
```diff
@@ -45,19 +49,22 @@ const StyledSliderInput = styled(SliderInput)`
   }
 `
 
+// TODO: fix slider losing focus after one keyboard left or right
```
Comment (code-arman) [inline src/components/ui/Slider.js:52]: for a future pr?

**Review by code-arman — APPROVED** [review]
Great work Quarantine! Beautiful PR, what's new?

I assume most of your TODO's are meant for later PR's but commented on to be sure you didn't forget anything.

 Also, 2 questions/comments about the behavior of this:
1) For seasonality, how does someone represent year long seasonality? 
2) When you press submit and you press the previous button, the stepper doesn't go back. I assume that's supposed to be the case because once you press submit, the screen will just go away?

File: src/components/ui/Slider.js (hunk 15 above)
Comment (Qrtn) [reply src/components/ui/Slider.js:52]: Yes!

File: src/components/ui/AddLocationPin.js (hunk 13 above)
Comment (Qrtn) [reply src/components/ui/AddLocationPin.js:14]: Everyone question 😜

File: src/components/ui/AddLocationPin.js (hunk 12 above)
Comment (Qrtn) [reply src/components/ui/AddLocationPin.js:5]: Yes, just wanted to get this code merged in. Shouldn’t make PRs that are too large, repeating a GLW mistake here LOL

File: src/components/form/ReportModal.js (hunk 11 above)
Comment (Qrtn) [reply src/components/form/ReportModal.js:30]: 😜

File: src/components/form/FormikStepper.js (hunk 7 above)
Comment (Qrtn) [reply src/components/form/FormikStepper.js:56]: Yes!

**Comment by Qrtn** [comment]
Great questions @code-arman and appreciate both of your guys’ reviews!

1. We should consider adding the no season checkbox to the form, or perhaps convert Jan to Dec to no season automatically. @SirajChokshi , you should take a look at the existing fallingfruit.org add location form. Note the no season checkbox and a few of the extra fields (date visited, etc)
2. Yes, haven’t hooked up the API yet, will handle that in a future PR.

**Pull request merged**